src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -173,11 +185,31 @@ def _update_command_table_from_extensions():
                         logger.warning("Unable to load extension '%s'. Use --debug for more information.", ext_name)
                         logger.debug(traceback.format_exc())
 
+        def _get_extension_suppress_filters(mod_loaders):
+            filter_func_name = 'module_suppress_extension'
+            return [getattr(m, filter_func_name) for m in mod_loaders if callable(getattr(m, filter_func_name, None))]
+
+        def _wrap_suppress_extension_func(func, ext):
+            """ Wrapper method to handle centralization of log messages for extension filters """
+            res = func(ext)
+            should_suppress = res
+            reason = "Use --debug for more information."
+            if isinstance(res, tuple):
+                should_suppress, reason = res
+            should_suppress = bool(should_suppress)
+            if should_suppress:
```

Comment:
The if statement will evaluate `should_suppress` to a bool anyway.
If a non-boolean value for `should_suppress` is supplied, this might be an indication that the user created a faulty suppression function that may have unintended results.
A check/warning for `should_suppress` not being a boolean will help catch these situations.

Reply:
It doesn't have to be a boolean.
Anything that is boolean-like would be fine.
So really I don't think a check is needed.

Even if in the suppression function, you returned an int equal to 1, it would be fine.
And, if you create a suppression function that returns a bad value, you would soon know when the function doesn't work.